src/URLHelper.hs

```diff
@@ -30,6 +32,17 @@ replaceIndexLinks = replace "href=\"/[^\"]*/index.html" P.takeDirectory
 replaceIndexURLs :: String -> Item String -> Compiler (Item String)
 replaceIndexURLs host = replace (host <> "/.*/index.html") P.takeDirectory
 
+-- | Replaces @/images/foo@ with @<host>/inages/foo@ for the given host
+replaceRelativeURLs :: String -> Item String -> Compiler (Item String)
+replaceRelativeURLs host = replace "=\"/.*\"" prependHost
+
+  where
+    prependHost = ("=\"" <> host </>) . (drop 3)
+
+-- | Repairs urls that have incorrectly had the given host prepended to them
+repairExternalURLs :: String -> Item String -> Compiler (Item String)
+repairExternalURLs host = replace (host <> "http") (drop (length host))
```

Comment:
the type of `replace` actually makes that _super_ difficult. I'm already using an implicit argument for the `Item String`, making it entirely point free ends up being pretty gross. I guess I could do `(drop $ length host)` instead though.